drivers/include/sdmmc/sdmmc.h

```diff
@@ -0,0 +1,1632 @@
+        uint32_t BLOCK_LEN_ERROR   :1;  /**< [29] SD and MMC */
+        uint32_t ADDRESS_ERROR     :1;  /**< [30] SD and MMC */
+        uint32_t OUT_OF_RANGE      :1;  /**< [31] SD/SDIO and MMC */
+    };
+    uint32_t value; /**< SD/MMC Card status as 32 bit value */
+} sdmmc_card_status_t;
+
+/** SD status register size */
+#define SDMMC_SD_STATUS_SIZE    (64)
+
+/**
+ * @brief   SD Status (SD Memory Card only)
+ *
+ * SD Status has a size of 512 bits and contains the status of SD Memory Card
+ * features. It is not supported by MMCs and SDIO Cards. In Combo Cards, it
+ * reflects the status of the SD Memory Card portion.
+ *
+ * Members are sorted by types to reduce the resuired RAM size.
+ *
+ * @see Physical Layer Simplified Specification Version 9.00, Section 4.10.2,
+ *      Table 4-44 [[sdcard.org](https://www.sdcard.org)]
+ */
+typedef struct {
+    uint32_t SIZE_OF_PROTECTED_AREA : 32;   /**< [479:448] Section 4.10.2.1 */
+    uint32_t SUS_ADDR : 22;                 /**< [367:346] Section 4.10.2.12, Table 4-57 */
+    uint32_t VSC_AU_SIZE : 10;              /**< [377:368] Section 4.10.2.11, Table 4-56 */
+    uint16_t SD_CARD_TYPE : 16;             /**< [495:480] Section 4.10.2, Table 4-44 */
+    uint16_t ERASE_SIZE : 16;               /**< [423:408] Section 4.10.2.5, Table 4-49 */
+    uint8_t  SPEED_CLASS : 8;               /**< [447:440] Section 4.10.2.2, Table 4-45 */
+    uint8_t  PERFORMANCE_MOVE : 8;          /**< [439:432] Section 4.10.2.3, Table 4-46 */
+    uint8_t  VIDEO_SPEED_CLASS : 8;         /**< [391:384] Section 4.10.2.10, Table 4-54 */
+    uint8_t  ERASE_TIMEOUT : 6;             /**< [407:402] Section 4.10.2.6, Table 4-50 */
+    uint8_t  ERASE_OFFSET : 2;              /**< [401:400] Section 4.10.2.7, Table 4-51 */
+    uint8_t  UHS_SPEED_GRADE : 4;           /**< [399:396] Section 4.10.2.8, Table 4-52 */
+    uint8_t  UHS_AU_SIZE : 4;               /**< [395:392] Section 4.10.2.9, Table 4-3 */
+    uint8_t  AU_SIZE : 4;                   /**< [431:428] Section 4.10.2.4, Table 4-47 */
+    uint8_t  DAT_BUS_WIDTH : 2;             /**< [511:510] Section 4.10.2, Table 4-44 */
+    uint8_t  SECURED_MODE : 1;              /**< [509] Section 4.10.2, Table 4-44 */
+} sdmmc_sd_status_t;
+
+/** @} */
+
+/**
+ * @name CID register (SD Memory and MMC Cards)
+ *
+ * @see Physical Layer Simplified Specification Version 9.00, Section 5.2,
+ *      Table 5.2 [[sdcard.org](https://www.sdcard.org)]
+ * @{
+ */
+
+/** CID register size in bytes (CID is 128 bit) */
+#define SDMMC_CID_REG_SIZE  (16)
+
+#define SDMMC_CID_OID_SIZE  (2)   /**< OID (OEM/Application ID) size in byte */
+#define SDMMC_CID_PNM_SIZE  (5)   /**< PNM (Product name) size in byte */
+
+/**
+ * @brief   CID register structure (SD Memory and MMC Cards)
+ *
+ * @see Physical Layer Simplified Specification Version 9.00, Section 5.2,
+ *      Table 5.2 [[sdcard.org](https://www.sdcard.org)]
+ */
+typedef struct {
```

Comment:
should that be a backed struct if it's supposed to be read like that from the SD card's memory?

Reply:
> should that be a packed struct

For some reason, I was assuming that the compiler options are set so that structs are always packed.

This data structure isn't read directly from memory since PNM it differs for SD Memory and MMC.